application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -188,17 +199,31 @@ private void deleteTag(@NotNull CommandInteraction event) {
      * @param actionVerb the verb describing the executed action, i.e. <i>edited</i> or
      *        <i>created</i>, will be displayed in the message send to the user
      * @param event the event to send messages with, it must have an {@code id} option set
+     * @param subcommand the executed subcommand
+     * @param newContent the new content of the tag (@{@link Nullable}: only if assigning new
+     *        content)
      */
     private void handleAction(@NotNull TagStatus requiredTagStatus,
             @NotNull Consumer<? super String> idAction, @NotNull String actionVerb,
-            @NotNull CommandInteraction event) {
+            @NotNull CommandInteraction event, @NotNull Subcommand subcommand,
+            @Nullable String newContent) {
+
         String id = Objects.requireNonNull(event.getOption(ID_OPTION)).getAsString();
         if (isWrongTagStatusAndHandle(requiredTagStatus, id, event)) {
             return;
         }
 
+        String previousContent = null;
+        if (subcommand == Subcommand.EDIT || subcommand == Subcommand.DELETE) {
+            previousContent = tagSystem.getTag(id).orElseThrow();
```

Comment:
The `orElseThrow()` is not necessarily safe. Although unlikely, imagine someone executes the delete command twice and then due to multithreading, one processes far enough before being paused, then the other executes fully and deletes, then the first branch continues and ends up calling `getTag(id).orElseThrow()`, crashing.

I think you should handle this case, silently ignore it and write a `debug` level message instead.

Reply:
done. see https://github.com/Together-Java/TJ-Bot/commit/460745ec1a67ec09f2ce15cdf35a6f6363626af5